src/freertos_drivers/common/SPI.hxx

```diff
@@ -54,6 +55,78 @@ public:
     /** Function point for the chip select assert and deassert methods */
     typedef void (*ChipSelectMethod)();
 
+    /** Method to transmit/receive the data.  Chip select will be asserted at
+     * start of the transfer and deasserted at the end of the transfer.
+     * @param msg message(s) to transact.
+     * @return bytes transfered upon success, -errno upon failure
+     */
+    __attribute__((optimize("-O3")))
+    int transfer_with_cs_assert(struct spi_ioc_transfer *msg)
+    {
+        csAssert();
+        int result = transfer(msg);
+        csDeassert();
+        return result;
+    }
+
+    /** Method to transmit/receive the data.  Chip select will be asserted at
+     * start of the transfer and deasserted at the end of the transfer.  This
+     * will always be a polled transaction no matter what.
+     * @param msg message(s) to transact.
+     * @return bytes transfered upon success, -errno upon failure
+     */
+    __attribute__((optimize("-O3")))
+    int transfer_with_cs_assert_polled(struct spi_ioc_transfer *msg)
+    {
+        csAssert();
+        int result = transfer_polled(msg);
+        csDeassert();
+        return result;
+    }
+
+    /** Conduct multiple message transfers with one stop at the end.
+     * @param msgs array of messages to transfer
+     * @param num number of messages to transfer
+     * @return total number of bytes transfered, -errno upon failure
+     */
+    __attribute__((optimize("-O3")))
+    int transfer_messages(struct spi_ioc_transfer *msgs, int num)
+    {
+        //HASSERT(num > 0);
+
+        int count = 0;
+        int result;
+
+        lock_.lock();
+        bus_lock();
+        for (int i = 0; i < num; ++i, ++msgs)
+        {
+            count += msgs->len;
+            csAssert();
+            result = transfer(msgs);
+            if (UNLIKELY(result < 0))
+            {
+                /* something bad happened, reset the bus and bail */
+                csDeassert();
+                bus_unlock();
+                lock_.unlock();
```

Comment:
nit: whenever you see "foo.unlock(); return;" sequence of commands, consider whether a RAII object would have made the code simpler, e.g. `OSMutexLock l(&lock);` at the beginning, then use just `return;` and let the destructor take care of unlocking.
Similar for portENTER_CRITICAL and portEXIT_CRITICAL pairs, using an AtomicHolder is a good alternative.
Same thing applies for break; when getting out of loops.